scripts/lint.js

```diff
@@ -1,42 +1,40 @@
 const chalk = require('chalk');
-const EslintCLI = require('eslint').CLIEngine;
-const eslintConfig = require('../config/eslint/eslintConfig');
 const isTypeScript = require('../lib/isTypeScript');
+const esLintCheck = require('../lib/runESLint').check;
 const prettierCheck = require('../lib/runPrettier').check;
 const runTsc = require('../lib/runTsc');
 const runTSLint = require('../lib/runTSLint');
 const args = require('../config/args').argv;
+const pathsToCheck = args.length > 0 ? args : undefined;
 
 (async () => {
   console.log(chalk.cyan('Linting'));
 
-  if (isTypeScript) {
-    try {
-      await runTsc();
-      await runTSLint();
-    } catch (e) {
-      console.log(e);
+  try {
+    if (isTypeScript) {
+      const hasPaths = typeof pathsToCheck !== 'undefined';
+      const pathsIncludeTS =
+        hasPaths &&
+        pathsToCheck.filter(
+          filePath => filePath.endsWith('.ts') || filePath.endsWith('.tsx'),
+        ).length > 0;
 
-      process.exit(1);
-    }
-  }
-
-  const cli = new EslintCLI({
-    baseConfig: eslintConfig,
-    useEslintrc: false,
-  });
+      await runTSLint(pathsToCheck);
 
-  const pathsToCheck = args.length === 0 ? ['.'] : args;
-
-  const formatter = cli.getFormatter();
-  console.log(chalk.gray(`eslint ${pathsToCheck.join(' ')}`));
-  const report = cli.executeOnFiles(pathsToCheck);
+      if (!hasPaths || pathsIncludeTS) {
```

Comment:
JSON files should probably trigger a `tsc` run too, since TypeScript treats them as typed files.

Reply:
Happy to add that, excluding obvious files like `package.json` and `package-lock.json` though right?